src/query/catalog/src/catalog/interface.rs

```diff
@@ -95,14 +96,21 @@ pub trait Catalog: DynClone + Send + Sync {
     /// Table.
 
     // Build a `Arc<dyn Table>` from `TableInfo`.
-    fn get_table_by_info(&self, table_info: &TableInfo) -> Result<Arc<dyn Table>>;
+    // If `ctx` is not None, then will get table schema field default values when init `dyn Table`.
+    fn get_table_by_info(
+        &self,
+        ctx: Option<Arc<dyn TableContext>>,
```

Comment:
`TableContext` argument is Option looks strange, it's hard to understand when we need it.

Reply:
Is there another way to do it instead of adding the TableContext to the function?

Reply:
How about add other API like `load_table`, this will never init default values of table schema, while `get_table` will init defaul values, use different API name to distinguish between different functions.

>> Is there another way to do it instead of adding the TableContext to the function?

Currently, `parse_exprs` need `TableContext` param, that is the root reason of this change.